dbcrossbarlib/src/drivers/postgres_shared/data_type.rs

```diff
@@ -0,0 +1,272 @@
+            DataType::Date => Ok(PgScalarDataType::Date),
+            DataType::Decimal => Ok(PgScalarDataType::Numeric),
+            DataType::Float32 => Ok(PgScalarDataType::Real),
+            DataType::Float64 => Ok(PgScalarDataType::DoublePrecision),
+            DataType::GeoJson => Ok(PgScalarDataType::Geometry(Srid::default())),
+            DataType::Int16 => Ok(PgScalarDataType::Smallint),
+            DataType::Int32 => Ok(PgScalarDataType::Int),
+            DataType::Int64 => Ok(PgScalarDataType::Bigint),
+            DataType::Json => Ok(PgScalarDataType::Jsonb),
+            DataType::Other(_) => Ok(PgScalarDataType::Text),
+            DataType::Text => Ok(PgScalarDataType::Text),
+            DataType::TimestampWithoutTimeZone => {
+                Ok(PgScalarDataType::TimestampWithoutTimeZone)
+            }
+            DataType::TimestampWithTimeZone => {
+                Ok(PgScalarDataType::TimestampWithTimeZone)
+            }
+            DataType::Uuid => Ok(PgScalarDataType::Uuid),
+        }
+    }
+
+    /// Convert this `PgDataType` to a portable `DataType`.
+    pub(crate) fn to_data_type(&self) -> Result<DataType> {
+        match self {
+            PgScalarDataType::Boolean => Ok(DataType::Bool),
+            PgScalarDataType::Date => Ok(DataType::Date),
+            PgScalarDataType::Numeric => Ok(DataType::Decimal),
+            PgScalarDataType::Real => Ok(DataType::Float32),
+            PgScalarDataType::DoublePrecision => Ok(DataType::Float64),
+            PgScalarDataType::Geometry(srid) if *srid == Srid::default() => {
+                Ok(DataType::GeoJson)
+            }
+            PgScalarDataType::Geometry(_srid) => Ok(DataType::Text),
+            PgScalarDataType::Smallint => Ok(DataType::Int16),
+            PgScalarDataType::Int => Ok(DataType::Int32),
+            PgScalarDataType::Bigint => Ok(DataType::Int64),
+            PgScalarDataType::Jsonb | PgScalarDataType::Json => Ok(DataType::Json),
+            PgScalarDataType::Text => Ok(DataType::Text),
+            PgScalarDataType::TimestampWithoutTimeZone => {
+                Ok(DataType::TimestampWithoutTimeZone)
+            }
+            PgScalarDataType::TimestampWithTimeZone => {
+                Ok(DataType::TimestampWithTimeZone)
+            }
+            PgScalarDataType::Uuid => Ok(DataType::Uuid),
+        }
+    }
+
+    /// See [this list of types and OIDs][types].
+    ///
+    /// [types]: https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_type.dat
+    pub(crate) fn oid(&self) -> Result<i32> {
+        match self {
+            PgScalarDataType::Boolean => Ok(16),
+            PgScalarDataType::Date => Ok(1082),
+            PgScalarDataType::Numeric => Ok(1700),
+            PgScalarDataType::Real => Ok(700),
+            PgScalarDataType::DoublePrecision => Ok(701),
+            PgScalarDataType::Geometry(_) => Err(format_err!(
+                "don't know the PostgreSQL OID for type `geometry`"
```

Comment:
it appears to vary, i think you will have to look it up every time:

server 1:

```
postgres=# SELECT oid, typname FROM pg_type where typname ~ 'geometry';
  oid   |      typname
--------+-------------------
 637731 | geometry
 637739 | _geometry
 638030 | geometry_dump
 638029 | _geometry_dump
 638555 | geometry_columns
 638554 | _geometry_columns
(6 rows)
```

server 2:

```
postgres=# SELECT oid, typname FROM pg_type where typname ~ 'geometry';
    oid    |      typname
-----------+-------------------
     17906 | geometry
     17914 | _geometry
     18185 | geometry_dump
     18184 | _geometry_dump
 129514304 | geometry_columns
 129514303 | _geometry_columns
(6 rows)
```

code like https://github.com/rgeo/activerecord-postgis-adapter/blob/master/lib/active_record/connection_adapters/postgis/oid/spatial.rb sets it dynamically

Reply:
Thank you for investigating! And gah, I was afraid of that. For now, I think I'll just punt on `geometry[]` array columns. We can always add these features later if we need them.